src/Book.java

```diff
@@ -0,0 +1,29 @@
+public class Book {
+    private String title;
+    private Author author;
+    private int publicationYear;
+
+
+    public Book(String title,Author author,int publicationYear) {
+        this.title = title;
+        this.author=author;
+        this.publicationYear = publicationYear;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public Author getAuthor() {
+        return author;
+    }
+
+    public int getPublicationYear() {
+        return getPublicationYear();
```

Comment:
Нужно возвращать значение свойства, а в твоем случае метод вызывает сам себя попадая в бесконечный цикл
```suggestion
        return publicationYear;
```